PokeAlarm/Events/QuestEvent.py

```diff
@@ -0,0 +1,77 @@
+# Standard Library Imports
+import datetime
+# 3rd Party Imports
+# Local Imports
+from PokeAlarm import Unknown
+from . import BaseEvent
+from PokeAlarm.Utils import get_gmaps_link, get_applemaps_link, \
+    get_waze_link, get_dist_as_str
+
+
+class QuestEvent(BaseEvent):
+    """ Event representing the discovery of a Quest. """
+
+    def __init__(self, data):
+        """ Creates a new Quest Event based on the given dict. """
+        super(QuestEvent, self).__init__('quests')
+        check_for_none = BaseEvent.check_for_none
+
+        # Identification
+        self.stop_id = data['pokestop_id']
+        self.stop_name = check_for_none(
+            str, data.get('pokestop_name') or data.get('name'),
+            Unknown.REGULAR)
+        self.stop_image = check_for_none(
+            str, data.get('pokestop_url') or data.get('url'), Unknown.REGULAR)
+
+        # Location
+        self.lat = float(data['latitude'])
+        self.lng = float(data['longitude'])
+
+        # Completed by Manager
+        self.distance = Unknown.SMALL
+        self.direction = Unknown.TINY
+
+        # Used to reject
+        self.name = self.stop_id
+        self.geofence = Unknown.REGULAR
+        self.custom_dts = {}
+
+        # Quest Details
+        self.quest = data['quest']
+        self.reward = data['reward']
+        self.expire_time = datetime.datetime.now().strftime("%d/%m/%Y 23:59")
```

Comment:
Where did we land on this? Seems like taking an argument with a default would be best?